lib/remote/configobjectutility.cpp

```diff
@@ -112,9 +118,24 @@ bool ConfigObjectUtility::CreateObject(const Type::Ptr& type, const String& full
 	String path = GetObjectConfigPath(type, fullName);
 	Utility::MkDirP(Utility::DirName(path), 0700);
 
-	if (Utility::PathExists(path)) {
-		errors->Add("Configuration file '" + path + "' already exists.");
-		return false;
+	{
+
+#ifdef _WIN32
+#define stat _stat
+#endif /* _WIN32 */
+
+		struct stat stats;
+
+		if (stat(path.CStr(), &stats)) {
+			if (errno != ENOENT)
+				BOOST_THROW_EXCEPTION(posix_error()
+				    << boost::errinfo_api_function("stat")
+				    << boost::errinfo_errno(errno)
+				    << boost::errinfo_file_name(path));
+		} else if (stats.st_size) {
+			errors->Add("Configuration file '" + path + "' already exists.");
+			return false;
+		}
```

Comment:
That will solve the problem with 0 byte files, but what if the file was scrambled with garbage? The real problem hiding here is that config files created from API requests are written "as is", without any temporary file & rename.